### ISSUE TITLE
V5/V6: UI freezes during model training — move .fit() off the GUI thread (QThread)

### ISSUE THREAD

**Issue by gbazad93** [gbazad93/PyQt6-DataWizard#4]
**Problem**

In `V5_MachineLearning/MachineLearning.py` (around line 226) and `V6_MachineLearningTabs/MachineLearningTabs.py` (around lines 280 and 369), the scikit-learn `.fit()` and `.predict()` calls run directly on the Qt GUI thread. There is no `QThread`, `QRunnable` or `QThreadPool` anywhere in either file, so the window stops repainting (Windows shows "Not Responding") for as long as training takes. It is very visible with a larger CSV or an SVR with expensive hyperparameters.

**Suggested approach**

- Move training into a small worker `QObject` with `finished(object)` / `error(str)` signals, and run it on a `QThread` (or use `QRunnable` + `QThreadPool`).
- While the worker runs: disable the Train button and show an indeterminate `QProgressBar` so the user gets feedback.
- Update the table, metrics labels and matplotlib canvas only in the main thread, from the `finished` slot. Never touch widgets from the worker thread.
- Do V5 first as a template, then apply the same pattern to the three model tabs in V6.

**Why this is a good first issue**

The change is contained to one method per file, and the repo already has a clear versioned structure to follow. Basic PyQt6 signal/slot knowledge is enough.

**Files**

- `V5_MachineLearning/MachineLearning.py`
- `V6_MachineLearningTabs/MachineLearningTabs.py`

**Comment by gbazad93** [comment]
Taking this one as a pairing task. Proposed approach so we agree before any code lands.

**1.** Start with `V5_MachineLearning/MachineLearning.py`: add a `TrainWorker(QObject)` with `finished(object)` and `error(str)` signals. Its `run()` does the `train_test_split` + `svr.fit()` + `predict()` and emits the fitted model and metrics as a plain dict.

**2.** In the window: create the `QThread`, call `worker.moveToThread(thread)`, connect `started -> run` and `finished -> quit/deleteLater`, and keep a reference so the thread is not garbage collected. Disable the Train button and start an indeterminate `QProgressBar` on launch; restore both in the `finished` and `error` slots.

**3.** All widget updates (table, metric labels, matplotlib canvas) happen in the main-thread slot only. Nothing in the worker touches Qt widgets.

**4.** Once V5 is reviewed, apply the identical pattern to the three model paths in `V6_MachineLearningTabs/MachineLearningTabs.py` (around lines 280 and 369) so the tabbed version stays consistent with V5.

**5.** Ship a regression test with it (ties into #5): assert the worker emits `finished` and that the Train button is re-enabled, running headless with `QT_QPA_PLATFORM=offscreen`.

Branch: `fix/v5-v6-qthread-training`. Commit trailer for the shared credit:

```
Co-authored-by: Bobby Azad <gbazad93@gmail.com>
```

Two things to settle before any code: `QThread` + worker object, or the lighter `QRunnable` + `QThreadPool`? And is cancellation (a Stop button) in scope here, or a follow-up issue?
